stackit/internal/services/sqlserverflex/instance/resource.go

```diff
@@ -0,0 +1,838 @@
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("instance_id"), idParts[1])...)
+	tflog.Info(ctx, "SQL Server Flex instance state imported")
+}
+
+func mapFields(ctx context.Context, resp *sqlserverflex.GetInstanceResponse, model *Model, flavor *flavorModel, storage *storageModel, options *optionsModel) error {
+	if resp == nil {
+		return fmt.Errorf("response input is nil")
+	}
+	if resp.Item == nil {
+		return fmt.Errorf("no instance provided")
+	}
+	if model == nil {
+		return fmt.Errorf("model input is nil")
+	}
+	instance := resp.Item
+
+	var instanceId string
+	if model.InstanceId.ValueString() != "" {
+		instanceId = model.InstanceId.ValueString()
+	} else if instance.Id != nil {
+		instanceId = *instance.Id
+	} else {
+		return fmt.Errorf("instance id not present")
+	}
+
+	var aclList basetypes.ListValue
+	var diags diag.Diagnostics
+	if instance.Acl == nil || instance.Acl.Items == nil {
+		aclList = types.ListNull(types.StringType)
+	} else {
+		respACL := *instance.Acl.Items
+		modelACL, err := utils.ListValuetoStringSlice(model.ACL)
+		if err != nil {
+			return err
+		}
+
+		reconciledACL := utils.ReconcileStringSlices(modelACL, respACL)
+
+		aclList, diags = types.ListValueFrom(ctx, types.StringType, reconciledACL)
+		if diags.HasError() {
+			return fmt.Errorf("mapping ACL: %w", core.DiagsToError(diags))
+		}
+	}
+
+	var flavorValues map[string]attr.Value
+	if instance.Flavor == nil {
+		flavorValues = map[string]attr.Value{
+			"id":          flavor.Id,
+			"description": flavor.Description,
+			"cpu":         flavor.CPU,
+			"ram":         flavor.RAM,
+		}
+	} else {
+		flavorValues = map[string]attr.Value{
+			"id":          types.StringValue(*instance.Flavor.Id),
+			"description": types.StringValue(*instance.Flavor.Description),
+			"cpu":         types.Int64PointerValue(instance.Flavor.Cpu),
+			"ram":         types.Int64PointerValue(instance.Flavor.Memory),
+		}
+	}
+	flavorObject, diags := types.ObjectValue(flavorTypes, flavorValues)
+	if diags.HasError() {
+		return fmt.Errorf("creating flavor: %w", core.DiagsToError(diags))
+	}
+
+	var storageValues map[string]attr.Value
+	if instance.Storage == nil {
+		storageValues = map[string]attr.Value{
+			"class": storage.Class,
+			"size":  storage.Size,
+		}
+	} else {
+		storageValues = map[string]attr.Value{
+			"class": types.StringValue(*instance.Storage.Class),
+			"size":  types.Int64PointerValue(instance.Storage.Size),
+		}
+	}
+	storageObject, diags := types.ObjectValue(storageTypes, storageValues)
+	if diags.HasError() {
+		return fmt.Errorf("creating storage: %w", core.DiagsToError(diags))
+	}
+
+	var optionsValues map[string]attr.Value
+	if instance.Options == nil {
+		optionsValues = map[string]attr.Value{
+			"edition":        options.Edition,
+			"retention_days": options.RetentionDays,
+		}
+	} else {
+		retentionDaysString := (*instance.Options)["retentionDays"]
```

Comment:
Is this field always present in the map?

Reply:
it may not be, I've updated the code